pkg/build/api/types.go

```diff
@@ -76,8 +76,72 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo
+}
+
+// BuildTriggerInfo contains information about why/who/what triggered a build.
+type BuildTriggerInfo struct {
+	// Details is a placeholder for  information about who triggered a build if
+	// it's either a user, a token or something else
+	Details BuildTriggerDetails
+
+	// CIJobDetails is a generic placeholder that should store information about
+	// jobs from CI plugins that integrate with openshift and triggered a
+	// specific build
+	CIDetails CIJobDetails
+
+	// GitWebHookInfo holds information about a build that was triggered by a
+	// gitwebhook including its refs, branch, last commit id, etc. It also stores
+	// the type of plugin that triggered it, if it was a generic one, github,
+	// bitbucket, etc.
+	GitWebHook GitWebHookInfo
```

Comment:
agreed, what i threw me off in the begging was the fact that card says we don't have info for generic webhooks  
https://trello.com/c/wozDHSJ2/717-8-track-reason-a-build-was-run-triggered
i did have them initially just like you suggested and then i did a more generic approach just to see how that goes